common/jack/metadata.h

```diff
@@ -204,19 +204,88 @@ int jack_set_property_change_callback (jack_client_t*             client,
                                        JackPropertyChangeCallback callback,
                                        void*                      arg);
 
-#ifdef __cplusplus
-} /* namespace */
-#endif
+/**
+ * ...
+ */
+extern const char* JACK_METADATA_CONNECTED;
 
 /**
- * @}
+ * The supported event types of an event port.
+ *
+ * This is a kludge around Jack only supporting MIDI, particularly for OSC.
+ * This property is a comma-separated list of event types, currently "MIDI" or
+ * "OSC".  If this contains "OSC", the port may carry OSC bundles (first byte
+ * '#') or OSC messages (first byte '/').  Note that the "status byte" of both
+ * OSC events is not a valid MIDI status byte, so MIDI clients that check the
+ * status byte will gracefully ignore OSC messages if the user makes an
+ * inappropriate connection.
  */
+extern const char* JACK_METADATA_EVENT_TYPES;
 
-extern const char* JACK_METADATA_PRETTY_NAME;
+/**
+ * ...
+ */
 extern const char* JACK_METADATA_HARDWARE;
-extern const char* JACK_METADATA_CONNECTED;
-extern const char* JACK_METADATA_PORT_GROUP;
-extern const char* JACK_METADATA_ICON_SMALL;
+
+/**
+ * ...
+ */
 extern const char* JACK_METADATA_ICON_LARGE;
 
+/**
+ * The name of the icon for the subject (typically client).
+ *
+ * This is used for looking up icons on the system, possibly with many sizes or
+ * themes.  Icons should be searched for according to the freedesktop Icon
+ *
+ * Theme Specification:
+ * http://standards.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html
+ */
+extern const char* JACK_METADATA_ICON_NAME;
+
+/**
+ * ...
+ */
+extern const char* JACK_METADATA_ICON_SMALL;
+
+/**
+ * Order for a port.
+ *
+ * This is used to specify the best order to show ports in user interfaces.
+ * The value MUST be an integer.  There are no other requirements, so there may
```

Comment:
A common practice with clients seems to be to set the property type to `http://www.w3.org/2001/XMLSchema#integer`. If this is a requirements or encouraged, it should be mentioned here.

Also, the value is encoded as a decimal numeric string, not as binary.